### PR TITLE
Tests/test tools base

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

Test coverage `test_tools/base.py`

```
Name                        Stmts   Miss  Cover
-----------------------------------------------
...
test_tools/abci_app.py         28      0   100%
test_tools/apis.py              5      0   100%
test_tools/base.py            138      0   100%
test_tools/common.py          114      0   100%
test_tools/integration.py     119     72    39%
test_tools/rounds.py          233      0   100%
```

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/test_tools/base.py [hunk 1]
```diff
@@ -143,9 +143,6 @@ def fast_forward_to_behaviour(
     ) -> None:
         """Fast forward the FSM to a behaviour."""
         next_behaviour = {s.behaviour_id: s for s in behaviour.behaviours}[behaviour_id]
-        assert (  # nosec
-            next_behaviour is not None
-        ), f"Behaviour {behaviour_id} not found"
```
Comment (Karrenbelt) [inline packages/valory/skills/abstract_round_abci/test_tools/base.py:148]: `behaviour.behaviours: Set[Type[BaseBehaviour]]`, that is to say one gets a `KeyError`, `None` is impossible.

File: packages/valory/skills/abstract_round_abci/tests/test_tools/test_base.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        test_instance.mock_contract_api_request(
+            contract_id, request_kwargs, response_kwargs
+        )
+
+
+def test_dummy_context_is_abstract_component() -> None:
+    """Test dummy context is abstract component"""
+
+    shared_state = SharedState(name="dummy_shared_state", skill_context=DummyContext())
+    assert shared_state.context.is_abstract_component
```
Comment (Karrenbelt) [inline packages/valory/skills/abstract_round_abci/tests/test_tools/test_base.py:206]: fair enough
https://github.com/valory-xyz/open-autonomy/blob/acaeb2aee6a038744014a5ead425a712afb4e36a/packages/valory/skills/abstract_round_abci/test_tools/base.py#L412-L415

File: packages/valory/skills/abstract_round_abci/tests/test_tools/test_base.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup class"""
+        cls.old_value = _MetaPayload.transaction_type_to_payload_cls.copy()  # type: ignore
+        _MetaPayload.transaction_type_to_payload_cls.clear()
+
+    @classmethod
+    def teardown_class(cls) -> None:
+        """Teardown class"""
+        _MetaPayload.transaction_type_to_payload_cls = cls.old_value  # type: ignore
```
Comment (0xArdi) [inline packages/valory/skills/abstract_round_abci/tests/test_tools/test_base.py:72]: Why is this necessary? Are you doing this to avoid interference?

Comment (Karrenbelt) [reply packages/valory/skills/abstract_round_abci/tests/test_tools/test_base.py:72]: yes. We do it all over the place. I made an issue out of this to be addressed during refactoring #1501

**Review by 0xArdi — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]
